compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeBuilder.java

```diff
@@ -3876,6 +3880,56 @@ public BLangNode transform(JoinClauseNode joinClauseNode) {
         return joinClause;
     }
 
+    @Override
+    public BLangNode transform(GroupByClauseNode groupByClauseNode) {
+        BLangGroupByClause groupByClause = (BLangGroupByClause) TreeBuilder.createGroupByClauseNode();
+        groupByClause.pos = getPosition(groupByClauseNode);
+
+        for (Node node : groupByClauseNode.groupingKey()) {
+            BLangGroupingKey groupingKeyNode = (BLangGroupingKey) TreeBuilder.createGroupingKeyNode();
+            groupingKeyNode.pos = getPosition(node);
+            if (node.kind() == SyntaxKind.SIMPLE_NAME_REFERENCE) {
+                groupingKeyNode.variableRef =
+                        (BLangSimpleVarRef) getGroupingKeySimpleVarRef((SimpleNameReferenceNode) node);
```

Comment:
Why can't we call transfrom method for `SimpleNameReferenceNode`

Reply:
transfrom method for SimpleNameReferenceNode creates a `BLangUserDefinedType` but what we need is a `BLangSimpleVarRef`.